### PR TITLE
Jaicp connectors refactoring and fixes, 0.7.0

### PR DESCRIPTION
**Pull request by Denire**
This pull request fixes working with asynchronous channels via JAICP with a bit of code cleanup and refactoring. With this pull request merged I think we're ready to release 0.7.0.

**The fix is**: gather a list of channels from JAICP project and initialise asynchronous webhook channels with proper urls to send responses.

### CODE REVIEW

File: channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt [hunk 1]
```diff
@@ -15,13 +21,59 @@ const val DEFAULT_PROXY_URL = "https://bot.jaicp.com"
...
+        }
+    }
+
+    private fun parseChannels(): List<Pair<JaicpChannelFactory, ChannelConfig>> {
+        val registeredChannels: List<ChannelConfig> = chatAdapterConnector.listChannels()
+        logger.info("Retrieved ${registeredChannels.size} channels configuration")
+
+        return channels.flatMap { factory ->
+            registeredChannels.mapNotNull {
+                if (factory.channelType.toUpperCase() == it.channelType) {
```
Comment (morfeusys) [inline channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt:65]: `equals` has `ignoreCase` option

File: channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt [hunk 2]
```diff
@@ -15,13 +21,59 @@ const val DEFAULT_PROXY_URL = "https://bot.jaicp.com"
...
+                    factory to it
+                } else
+                    null
+            }
+        }
+    }
+
+    abstract fun registerChannel(channel: JaicpBotChannel, channelConfig: ChannelConfig)
+
+    protected fun getChannelProxyUrl(config: ChannelConfig) =
```
Comment (morfeusys) [inline channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt:75]: Nobody overrides this

**Review by morfeusys — APPROVED** [review]

**Review by nikvoloshin — APPROVED** [review]

File: channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt (hunk 1 above)
Comment (Denire) [reply channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt:65]: Thanks! I've forgotten it.

File: channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt (hunk 2 above)
Comment (Denire) [reply channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt:75]: It doesn't, but another subclass (JaicpPollingConnector) uses it to set proper url

**Pull request merged**